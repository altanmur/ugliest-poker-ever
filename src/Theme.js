import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

export const theme = {
  colors: {
    indigo: '#4A0968',
    blackcurrant: '#27172F',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
