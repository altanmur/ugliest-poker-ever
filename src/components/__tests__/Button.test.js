import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Button from '../Button';
import { theme } from '../../Theme';

describe('Button Component', () => {
  it('should render correctly', () => {
    const component = shallow(<Button theme={theme} />);

    expect(component).toMatchSnapshot();
  });
});
