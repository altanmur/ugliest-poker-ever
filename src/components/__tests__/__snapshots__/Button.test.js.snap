// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: 2px solid #a8d90d;
  border-radius: 8px;
  background: linear-gradient(0deg,#308a00 1%,#c3e80e 80%);
  color: #1b2f04;
  cursor: pointer;
  font-weight: bold;
  margin: 10px 0;
  outline: none;
  padding: 5px 30px;
}

<button
  className="c0"
/>
`;
